src/rust/rust-proto/tests/test_utils/strategies.rs

```diff
@@ -1236,12 +1217,87 @@ pub mod analyzer_sdk {
         }
     }
 
-    pub fn updates() -> impl Strategy<Value = Update> {
+    pub fn updates() -> impl Strategy<Value = native::Update> {
+        prop_oneof![
+            string_property_updates().prop_map(native::Update::StringProperty),
+            uint_64_property_updates().prop_map(native::Update::Uint64Property),
+            int_64_property_updates().prop_map(native::Update::Int64Property),
+            edge_updates().prop_map(native::Update::Edge),
+        ]
+    }
+
+    pub fn string_operations() -> impl Strategy<Value = StringOperation> {
+        prop_oneof![
+            Just(StringOperation::Has),
+            Just(StringOperation::Equal),
+            Just(StringOperation::Contains),
+            Just(StringOperation::Regex),
+        ]
+    }
+
+    prop_compose! {
+        pub fn string_filters()(
+            operation in string_operations(),
+            value in string_not_empty(),
+            negated in any::<bool>(),
+        ) -> StringFilter {
+            StringFilter {
+                operation,
+                value,
+                negated,
+            }
+        }
+    }
+
+    pub fn int_operations() -> impl Strategy<Value = IntOperation> {
+        prop_oneof![
+            Just(IntOperation::Has),
+            Just(IntOperation::Equal),
+            Just(IntOperation::LessThan),
+            Just(IntOperation::LessThanOrEqual),
+            Just(IntOperation::GreaterThan),
+            Just(IntOperation::GreaterThanOrEqual),
+        ]
+    }
+
+    prop_compose! {
+        pub fn int_filters()(
+            operation in int_operations(),
+            value in any::<i64>(),
+            negated in any::<bool>(),
+        ) -> IntFilter {
+            IntFilter {
+                operation,
+                value,
+                negated,
+            }
+        }
+    }
+
+    pub fn uid_operations() -> impl Strategy<Value = UidOperation> {
+        prop_oneof![Just(UidOperation::Equal),]
+    }
+
+    prop_compose! {
+        pub fn uid_filters()(
+            operation in uid_operations(),
+            value in common::uids(),
+        ) -> UidFilter {
+            UidFilter {
+                operation,
+                value,
+            }
+        }
+    }
+
+    pub fn execution_results() -> impl Strategy<Value = native::ExecutionResult> {
         prop_oneof![
-            string_property_updates().prop_map(Update::StringProperty),
-            uint_64_property_updates().prop_map(Update::Uint64Property),
-            int_64_property_updates().prop_map(Update::Int64Property),
-            edge_updates().prop_map(Update::Edge),
+            Just(native::ExecutionResult::ExecutionMiss(
+                native::ExecutionMiss {}
+            )),
+            // TODO wimax Nov 10 2022: Flesh this out to also include ExecutionHit.
+            // Currently those are only tested in python, not rust.
+            // Just(native::ExecutionResult::ExecutionHit(native::ExecutionHit{})),
```

Comment:
unfortunately it'd be a lot of proptest boilerplate that I don't think is the right use of my time given Certain Timing Constraints